backend/tournesol/tests/test_text_search.py

```diff
@@ -0,0 +1,712 @@
+        self.assertEqual(response.data["results"][0]["uid"], relevant_entity.uid)
+        self.assertEqual(response.data["results"][1]["uid"], less_relevant_entity.uid)
+
+    def test_public_recommendations_multiple_matches(self):
+        """
+        If the word appears twice in a field, it should give a higher
+        score than if it appears only once.
+        """
+        query = "onion"
+        relevant_field = "how to cook an onion cake with onions"
+        less_relevant_field = "how to cook an onion cake"
+        unrelated_field = "how to build a house"
+
+        # Check that the relevance compensates for the lower score
+        relevant_entity = self._create_rated_entity(field1=relevant_field, criteria_score=1.0)
+        less_relevant_entity = self._create_rated_entity(
+            field1=less_relevant_field,
+            criteria_score=1.1
+        )
+        self._create_rated_entity(field1=unrelated_field, criteria_score=1.2)
+
+        response = self.client.get(
+            f"/users/{self.user.username}/recommendations/{self.poll.name}"
+            f"?search={query}&metadata[language]={self.language}",
+            format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 2)
+        self.assertEqual(response.data["results"][0]["uid"], relevant_entity.uid)
+        self.assertEqual(response.data["results"][1]["uid"], less_relevant_entity.uid)
+
+    def test_private_recommendations_multiple_matches(self):
+        """
+        If the word appears twice in a field, it should give a higher
+        score than if it appears only once.
+        """
+        query = "onion"
+        relevant_field = "how to cook an onion cake with onions"
+        less_relevant_field = "how to cook an onion cake"
+        unrelated_field = "how to build a house"
+
+        # Check that the relevance compensates for the lower score
+        relevant_entity = self._create_rated_entity(field1=relevant_field, criteria_score=1.0)
+        less_relevant_entity = self._create_rated_entity(
+            field1=less_relevant_field,
+            criteria_score=1.1
+        )
+        self._create_rated_entity(field1=unrelated_field, criteria_score=1.2)
+
+        self.client.force_authenticate(self.user)
+        response = self.client.get(
+            f"/users/me/recommendations/{self.poll.name}"
+            f"?search={query}&metadata[language]={self.language}",
+            format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 2)
+        self.assertEqual(response.data["results"][0]["uid"], relevant_entity.uid)
+        self.assertEqual(response.data["results"][1]["uid"], less_relevant_entity.uid)
+
+    def test_multiple_words(self):
+        """
+        Test with multiple words in the query.
+        Leave the freedom to accept or reject an entity for

```

Comment:
Leave the freedom to who? : P

If we are delegating a decision to an underlying system, we can make it more explicit by writing the name of the system.

Example: The PostgreSQL full-text search can discard results that seem less relevant to it, when only one word matches for instance. 

If it was an unexpected behaviour that we cannot configure, we can mention it too.

Example: Due to a limitation of the PostgreSQL full-text search... etc.

Reply:
Now that I make a pre-filtering ("qs.filter(search_vector=search_query)" in filter_search), all the query words (except stop words) have to appear in the entity in order to not get filtered. 
It's not obvious if it is the desired behavior. But I'll thus consider that only 1 result should be returned.